pyccel/ast/operators.py

```diff
@@ -197,27 +197,25 @@ class PyccelUnaryOperator(PyccelOperator):
     def __init__(self, arg):
         super().__init__(arg)
 
-    def _set_dtype(self):
+    @staticmethod
+    def _calculate_dtype(_args):
```

Comment:
Have you thought about using [tuple unpacking](https://www.geeksforgeeks.org/packing-and-unpacking-arguments-in-python/) (i.e. `*_args`) here instead of passing `_args`?

Reply:
I ll try that